plutus-core/plutus-core/src/PlutusCore/Parser/Builtin.hs

```diff
@@ -145,7 +145,7 @@ conBLS12_381_G2_Element = do
 -- | Parser for constants of the given type.
 constantOf :: ExpectParens -> DefaultUni (Esc a) -> Parser a
 constantOf expectParens uni =
-  case uni of
+  try (trailingWhitespace . inParens $ constantOf ExpectParensNo uni) <|> case uni of
```

Comment:
Also, the prettyprinter still prints list elements without parentheses. I suppose we have to choose one way or the other, so that seems fine.